### PR TITLE
Clean up limel-list and limel-menu-list a bit

### PR DESCRIPTION
**Pull request by adrianschmidt**
This is the second step in splitting all menu related functionality out of limel-list.

The first step was #1416

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/menu-list/menu-list-renderer.tsx
Comment (hannahu) [inline src/components/menu-list/menu-list-renderer.tsx:59]: Some of the changes to this file belongs to the third commit right? Since they are menu related.

File: src/components/menu-list/menu-list.tsx
Comment (hannahu) [inline src/components/menu-list/menu-list.tsx:56]: Same here

File: src/components/menu-list/menu-list.types.ts
Comment (hannahu) [inline src/components/menu-list/menu-list.types.ts:6]: Same here

File: src/components/menu/menu.types.ts
Comment (hannahu) [inline src/components/menu/menu.types.ts:12]: Shouldn't we change all list item to _menu item_ here?

File: src/components/menu/menu.types.ts
Comment (hannahu) [inline src/components/menu/menu.types.ts:49]: Can we have actions on a menu item?

File: src/components/menu/menu.types.ts
Comment (adrianschmidt) [reply src/components/menu/menu.types.ts:12]: True! Good catch 👍

File: src/components/menu/menu.types.ts
Comment (adrianschmidt) [reply src/components/menu/menu.types.ts:49]: I guess it would make sense if we had sub-menus, but I don't think we do, so no, these should go. They'll be removed in a later commit though (possibly in a later PR even).

File: src/components/menu-list/menu-list-renderer.tsx
Comment (adrianschmidt) [reply src/components/menu-list/menu-list-renderer.tsx:59]: Hmm… yeah, that's weird… I must have mixed something up while juggling different fixups or something… Thanks for the catch! 🙏

Comment (adrianschmidt) [reply src/components/menu-list/menu-list-renderer.tsx:59]: ⚡

File: src/components/menu-list/menu-list.tsx
Comment (adrianschmidt) [reply src/components/menu-list/menu-list.tsx:56]: ⚡

File: src/components/menu-list/menu-list.types.ts
Comment (adrianschmidt) [reply src/components/menu-list/menu-list.types.ts:6]: ⚡

**Review by hannahu — DISMISSED** [review]

**Comment by hannahu** [comment]
Approved! Tests failing because of the removal of the actions on the menu item interface :)

File: src/components/menu/menu.types.ts
Comment (adrianschmidt) [reply src/components/menu/menu.types.ts:49]: Removing this made other things fail, so I'm moving that over to my step 3 branch. Whether we merge this first, or wait for that branch to merge into here, I'm good either way, I just want to keep this point as a fixed green build point that I won't accidentally fumble away 😂

Comment (hannahu) [reply src/components/menu/menu.types.ts:49]: Okey! :P

**Review by hannahu — APPROVED** [review]

**Pull request merged**
